src/Modules/UngradedModules/FreeMod.jl

```diff
@@ -153,13 +153,16 @@ end
 @doc raw"""
     rank(F::FreeMod)
     number_of_generators(F::AbstractFreeMod)
-    dim(F::AbstractFreeMod)
+    dim(F::AbstractFreeMod{T}) where T <: FieldElem
+    vector_space_dim(F::AbstractFreeMod{T}) where T <: FieldElem
 
 Return the rank of `F`.
 """
-dim(F::AbstractFreeMod) = rank(F)
 rank(F::FreeMod) = F.n
+dim(F::AbstractFreeMod{T}) where T <: FieldElem = rank(F)
+dim(kk::Union{Field, ModuleFP}) = error("The value of `dim` is ambiguous for $(typeof(kk)), see `krull_dim`, `vector_space_dim` or `rank`.")
```

Comment:
Defining `dim` like that for `Field`, both if which are from AA, is type piracy. Do we really need this method? If we don't define it we'll simply get a MethodError I think?

Reply:
I wanted to give a more descriptive error message, but I suppose this can be moved to AA

Reply:
Only the `Field` mehod -- but really I don't see the next. Just change this to
```suggestion
dim(M::ModuleFP) = error("The value of `dim` is ambiguous for $(typeof(M)), use `krull_dim`, `vector_space_dim` or `rank` instead")
```
and then let's move on.